File: frontend/src/api/spaceController.ts
// @ts-ignore
/* eslint-disable */
import request from "@/ui-backend/utils";

/** 此处后端没有提供注释 GET /backend/space/ */
export async function getSpaceVoById(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getSpaceVOByIdParams,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseSpaceVO>("/backend/space/", {
    method: "GET",
    params: {
      ...params,
    },
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /backend/space/ */
export async function addSpace(
  body: API.SpaceAddRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseBoolean>("/backend/space/", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 DELETE /backend/space/ */
export async function deleteSpace(
  body: API.SpaceDeleteRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseBoolean>("/backend/space/", {
    method: "DELETE",
    headers: {
      "Content-Type": "application/json",
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /backend/space/list_user */
export async function getSpaceVoListByUserId(
  body: API.SpaceByUserIdRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseListSpaceVO>("/backend/space/list_user", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 GET /backend/space/list/level */
export async function getSpaceListLevel(options?: { [key: string]: any }) {
  return request<API.BaseResponseListSpaceLevel>("/backend/space/list/level", {
    method: "GET",
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 POST /backend/space/list/page */
export async function getSpaceVoListByPage(
  body: API.SpaceQueryRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponsePageSpaceVO>("/backend/space/list/page", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    data: body,
    ...(options || {}),
  });
}

/** 此处后端没有提供注释 PUT /backend/space/update */
export async function updateSpace(
  body: API.SpaceUpdateRequest,
  options?: { [key: string]: any }
) {
  return request<API.BaseResponseBoolean>("/backend/space/update", {
    method: "PUT",
    headers: {
      "Content-Type": "application/json",
    },
    data: body,
    ...(options || {}),
  });
}

File: src/main/java/org/example/blogsakura/interfaces/controller/SpaceController.java
package org.example.blogsakura.interfaces.controller;

import com.mybatisflex.core.paginate.Page;
import jakarta.annotation.Resource;
import jakarta.servlet.http.HttpServletRequest;
import lombok.extern.slf4j.Slf4j;
import org.example.blogsakura.infrastruct.annotation.AuthCheck;
import org.example.blogsakura.infrastruct.common.BaseResponse;
import org.example.blogsakura.infrastruct.common.ResultUtils;
import org.example.blogsakura.domain.user.constant.UserConstant;
import org.example.blogsakura.interfaces.vo.space.SpaceVO;
import org.example.blogsakura.interfaces.dto.space.*;
import org.springframework.web.bind.annotation.*;
import org.example.blogsakura.application.service.SpaceApplicationService;

import java.util.List;

/**
 * 空间 控制层。
 * TODO: 注意，用户只能创建一个空间，所谓空间创始人可以是用户，但只能创建一个空间，只有管理员可以多次创建空间
 *
 * @author <a href="https://github.com/liuweixu">liuweixu</a>
 */
@RestController
@RequestMapping("/backend/space")
@Slf4j
public class SpaceController {

    @Resource
    private SpaceApplicationService spaceApplicationService;


    /**
     * 创建空间（所有人都可以使用）。
     *
     * @param spaceAddRequest 空间创建请求
     * @return {@code true} 保存成功，{@code false} 保存失败
     */
    @PostMapping("/")
    public BaseResponse<Boolean> addSpace(@RequestBody SpaceAddRequest spaceAddRequest,
                                          HttpServletRequest request) {
        return ResultUtils.success(spaceApplicationService.addSpace(spaceAddRequest, request));
    }

    /**
     * 获取空间信息
     *
     * @param id
     * @param request
     * @return
     */
    @GetMapping("/")
    public BaseResponse<SpaceVO> getSpaceVOById(@RequestParam Long id, HttpServletRequest request) {
        return ResultUtils.success(spaceApplicationService.getSpaceVOById(id, request));
    }


    /**
     * 根据主键删除空间。针对管理员或者用户
     *
     * @param spaceDeleteRequest 主键
     * @return {@code true} 删除成功，{@code false} 删除失败
     */
    @DeleteMapping("/")
    public BaseResponse<Boolean> deleteSpace(@RequestBody SpaceDeleteRequest spaceDeleteRequest, HttpServletRequest request) {
        return ResultUtils.success(spaceApplicationService.deleteSpace(spaceDeleteRequest, request));
    }

    /**
     * 获取封装后的空间信息分页
     *
     * @param spaceQueryRequest
     * @param request
     * @return
     */
    @PostMapping("list/page")
    public BaseResponse<Page<SpaceVO>> getSpaceVOListByPage(@RequestBody SpaceQueryRequest spaceQueryRequest,
                                                            HttpServletRequest request) {
        return ResultUtils.success(spaceApplicationService.getSpaceVOListByPage(spaceQueryRequest, request));
    }


    /**
     * 仅限管理员更新更新空间。
     *
     * @param spaceUpdateRequest 空间更新请求
     * @return {@code true} 更新成功，{@code false} 更新失败
     */
    @PutMapping("update")
    @AuthCheck(mustRole = UserConstant.ADMIN_ROLE)
    public BaseResponse<Boolean> updateSpace(@RequestBody SpaceUpdateRequest spaceUpdateRequest) {
        return ResultUtils.success(spaceApplicationService.updateSpace(spaceUpdateRequest));
    }

    /**
     * 查询空间级别列表
     *
     * @return
     */
    @GetMapping("/list/level")
    public BaseResponse<List<SpaceLevel>> getSpaceListLevel() {
        return ResultUtils.success(spaceApplicationService.getSpaceListLevel());
    }

    /**
     * 从用户id获取相应的空间列表信息
     *
     * @param spaceByUserIdRequest
     * @return
     */
    @PostMapping("/list_user")
    public BaseResponse<List<SpaceVO>> getSpaceVOListByUserId(@RequestBody SpaceByUserIdRequest spaceByUserIdRequest) {
        return ResultUtils.success(spaceApplicationService.getSpaceVOListByUserId(spaceByUserIdRequest));
    }
}
